### PR TITLE
Only duplicate records from archive to writable during online_delete.

### PR DESCRIPTION
**Pull request by mtrippled**
Log duration of nodestore reads.
Include nodestore counters in perf_log output.

Remove gratuitous nodestore activity counting.

Report initial sync duration in server_info and perfLog.
Report state_accounting in perfLog.
Make state_accounting durations more accurate.

Parallel ledger loader.

Config parameter to load ledgers on start.

### Type of Change

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Refactor (non-breaking change that only restructures code)
- [ ] Tests (You added tests for code that already exists, or your new feature included in this PR)
- [ ] Documentation Updates
- [ ] Release

### CODE REVIEW

File: src/ripple/app/ledger/Ledger.cpp [hunk 1]
```diff
@@ -773,7 +773,10 @@ Ledger::walkLedger(beast::Journal j) const
     }
     else
     {
-        stateMap_->walkMap(missingNodes1, 32);
```
Comment (nbougalis) [inline src/ripple/app/ledger/Ledger.cpp:776]: Generally, using `walkMap` is the wrong thing, but let's not worry about that for this PR.

File: src/ripple/app/main/Main.cpp [hunk 2]
```diff
@@ -626,23 +626,16 @@ run(int argc, char** argv)
             config->START_UP = Config::REPLAY;
         else
             config->START_UP = Config::LOAD;
-    }
-    else if (vm.count("ledgerfile"))
+    } else if (vm.count("ledgerfile"))
```
Comment (nbougalis) [inline src/ripple/app/main/Main.cpp:629]: Fails `clang-format`

File: src/ripple/app/main/Main.cpp [hunk 3]
```diff
@@ -626,23 +626,16 @@ run(int argc, char** argv)
...
     {
         config->START_LEDGER = vm["ledgerfile"].as<std::string>();
         config->START_UP = Config::LOAD_FILE;
-    }
-    else if (vm.count("load"))
+    } else if (vm.count("load") || config->FAST_LOAD)
```
Comment (nbougalis) [inline src/ripple/app/main/Main.cpp:633]: Fails `clang-format`

**Review by nbougalis — COMMENTED** [review]
Left a couple of small comments related to formatting. Looks fine.

Please reword the commit:

```
Parallel ledger loader & I/O performance improvements:
 
- Only duplicate records from archive to writable during online_delete.
- Log duration of nodestore reads.
- Include nodestore counters in perf_log output.
- Remove gratuitous nodestore activity counting.
- Report initial sync duration in server_info and perfLog.
- Report state_accounting in perfLog.
- Make state_accounting durations more accurate.
- Parallel ledger loader.
- Config parameter to load ledgers on start.
```

**Comment by mtrippled** [comment]
@nbougalis clang-formatted, comment fixed, unit test fixed

**Comment by scottschurr** [comment]
I'm still working on this review, but the levelization CI failure really needs to be addressed before this pull request can be merged.

The levelization needs a longer term fix, but you can temporarily hack the CI to silence the complaint by cherry-picking this commit: 

https://github.com/scottschurr/rippled/commit/93a0377781b5656a9d7121463e1a110ccbf12da5

That commit does nothing to address the real problem, it only masks the problem.  Any real solutions I've come up with are more extensive than I think want to go into this commit.  But maybe someone has a better idea than I do.

**Comment by ximinez** [comment]
Data point: I had timed sync on some Windows release builds last week and over the weekend to about 1, 1.5, and 2.5 hours. With `fast-load` set the sync time was reduced to about 0.5 hours.

File: src/ripple/shamap/impl/SHAMapDelta.cpp [hunk 4]
```diff
@@ -286,4 +290,81 @@ SHAMap::walkMap(std::vector<SHAMapMissingNode>& missingNodes, int maxMissing)
...
+    {
+        auto const& child = topChildren[rootChildIndex];
+        if (!child || !child->isInner())
+            continue;
+
+        nodeStacks[rootChildIndex].push(
+            std::static_pointer_cast<SHAMapInnerNode>(child));
+
+        JLOG(journal_.debug()) << "starting worker " << rootChildIndex;
+        std::mutex m;
```
Comment (scottschurr) [inline src/ripple/shamap/impl/SHAMapDelta.cpp:327]: This `mutex` needs to be in the outermost scope.  We need `m`'s destructor not to run until after all the calls to `join()` have completed.

**Review by scottschurr — COMMENTED** [review]
I suspect this commit is already on its way into develop.  But I think the comments may still be useful so they can be considered for follow-on patches.

**Comment by ximinez** [comment]
This branch (based on @seelabs's #4063 branch) fixes the levelization issue with ripple/basics dependent on ripple/nodestore. https://github.com/ximinez/rippled/commits/perflog-levelization

Rather than adding a dependency, I use a callback defined in `Application`. To enable that same callback in the unit tests, I made it a `static` function, but it could be inline if we don't care about that.

**Pull request closed without merging**